tmva/sofie/src/RModel.cxx

```diff
@@ -322,29 +331,42 @@ void RModel::Initialize(int batchSize, bool verbose) {
 }
 
 void RModel::GenerateInitializedTensorInfo() {
+    if (!fInitializedTensors.empty())
+      fGC += "// initialized tensors\n";
     for (auto& i: fInitializedTensors) {
-        if (i.second.type() == ETensorType::FLOAT) {
-            size_t length = 1;
-            for (auto & dim: i.second.shape()) {
-                length *= dim;
+
+         size_t length = ConvertShapeToLength(i.second.shape());
+            // in case we are not using weight files or for tensor created from Constant operator
+         if (!fUseWeightFile || i.second.IsConstantTensor() ) {
+            //std::cout << "write tensor " << i.first << std::endl;
+            std::stringstream strs;
+            if (i.second.type() == ETensorType::FLOAT) {
+               strs << "float tensor_" << i.first << "[" << length << "] = {";
+               float const *data = i.second.data<float>();
+               for (size_t idx = 0; idx < length; idx++) {
+                  strs << std::setprecision(std::numeric_limits<float>::max_digits10) << data[idx];
```

Comment:
a general comment: should the fp numbers be written in hex format not to loose any precision?